sfall/LoadGameHook.cpp

```diff
@@ -276,6 +277,45 @@ static void __declspec(naked) LoadGame() {
 		retn;
 	}
 }
+
+void ReadExtraGameMsgFiles()
+{
+	int read;
+	std::string names;
+
+	names.resize(256);
+
+	while ((read = GetPrivateProfileStringA("Misc", "ExtraGameMsgFileList", "",
+		(LPSTR)names.data(), names.size(), ".\\ddraw.ini")) == names.size() - 1)
+		names.resize(names.size() + 256);
+
+	if (names.empty())
+		return;
+
+	names.resize(names.find_first_of('\0'));
+	names.append(",");
+
+	int begin = 0;
+	int end;
+	int length;
+
+	while ((end = names.find_first_of(',', begin)) != std::string::npos)
+	{
+		length = end - begin;
+
+		if (length > 0)
+		{
+			std::string path = "game\\" + names.substr(begin, length) + ".msg";
```

Comment:
Ugh. Right. When I load a game, the files aren't read, since I placed the reading code in a function that gets called only when a new game is started (still, all the scripts are executed properly). That's a bug.

At first I wanted to read the files in the **DllMain2** function, where majority of ini settings are read and interpreted. But - for some reason - the files wouldn't parse there.

I suspect that wheter a file access succeeds or fails depends on the contents of linked **sPath** struct instances pointed at by **_paths** address. Nevertheless, I don't know how and when it changes. Do you know how does it work?